whylog/assistant/spans_finding.py

```diff
@@ -0,0 +1,107 @@
+"""
+Methods responsible for finding spans (potential params) for given text.
+Some heuristic included.
+"""
+
+import re
+from datetime import datetime
+
+from dateutil.parser import parse as date_parse
+
+from whylog.assistant.const import DataType, DateParams
+from whylog.assistant.regex_assistant.regex import DateFromFutureError, create_date_regex
+from whylog.assistant.span import Span, not_overlapping_spans, sort_as_date
+
+
+def find_date_spans(text, regexes=None):
+    regexes = regexes or {}
+
+    forced_date_spans = _find_date_spans_by_force(text)
+    regex_date_spans = _find_date_spans_by_regex(regexes, text)
+
+    unique_spans = forced_date_spans.union(regex_date_spans)
+
+    ordered_date_spans = sort_as_date(unique_spans)
+
+    best_spans = not_overlapping_spans(ordered_date_spans)
+
+    return best_spans
+
+
+def _find_date_spans_by_regex(regexes, text):
+    return _find_spans_by_regex(
+        regexes,
+        text,
+        pattern_creator=create_date_regex,
+        converter=DataType.DATE
+    )
+
+
+def _find_date_spans_by_force(text):
+    date_spans = set()
+
+    proper_starts, proper_ends = _possible_span_starts_and_ends(text)
+    proper_ends.reverse()
+
+    date_now = datetime.now()
+    not_overlapping_start = 0
+
+    for start in proper_starts:
+        if start < not_overlapping_start:
+            continue
+        for end in proper_ends:
+            if end < start + DateParams.MIN_DATE_LENGTH:
+                break
+            try:
+                date_text = text[start:end]
+                date = date_parse(date_text)
+                if date_now.year - date.year < -1:
+                    raise DateFromFutureError(date, date_text)
+            except (ValueError, DateFromFutureError):
+                # TODO: how to write it without bad looking 'pass'?
```

Comment:
There is no need to inform user about this Error. This error is thrown by date_parse in "date = date_parse(date_text)".  It means that date_text is not a date. It is normal and common situation while searching for a date in text. date_parse is alias for dateutil.parser.parse.